invertedai/large/drive.py

```diff
@@ -0,0 +1,188 @@
+    get_infractions: bool = False,
+    random_seed: Optional[int] = None,
+    api_model_version: Optional[str] = None,
+    single_call_agent_limit: Optional[int] = None,
+    async_api_calls: bool = True
+) -> DriveResponse:
+    """
+    A utility function to drive more than the normal capacity of agents in a call to :func:`drive`.
+    The agents are inserted into a quadtree structure and :func:`drive` is then called on each
+    region represented by a leaf node of the quadtree. Agents near this region are included in the
+    :func:`drive` calls to ensure the agents behave properly locally. The quadtree is constructed 
+    during each call to this utility function to maintain statelessness.
+
+    Parameters
+    ----------
+    location:
+        Location name in IAI format.
+
+    agent_states:
+        Current states of all agents.
+        The state must include x: [float], y: [float] coordinate in meters
+        orientation: [float] in radians with 0 pointing along x and pi/2 pointing along y and
+        speed: [float] in m/s.
+
+    agent_attributes:
+        Static attributes of all agents.
+        List of agent attributes. Each agent requires, length: [float]
+        width: [float] and rear_axis_offset: [float] all in meters. agent_type: [str],
+        currently supports 'car' and 'pedestrian'.
+        waypoint: optional [Point], the target waypoint of the agent.
+
+    recurrent_states:
+        Recurrent states for all agents, obtained from the previous call to
+        :func:`drive` or :func:`initialize` or the large equivalents.
+
+    traffic_lights_states:
+       If the location contains traffic lights within the supported area,
+       their current state should be provided here. Any traffic light for which no
+       state is provided will have a state generated by iai.
+
+    light_recurrent_states:
+       Light recurrent states for all agents, obtained from the previous call to
+        :func:`drive` or :func:`initialize` or the large equivalents.
+       Specifies the state and time remaining for each light group in the map.
+       If manual control of individual traffic lights is desired, modify the relevant state(s) 
+       in traffic_lights_states, then pass in light_recurrent_states as usual.
+
+    get_infractions:
+        Whether to check predicted agent states for infractions.
+        This introduces some overhead, but it should be relatively small.
+
+    random_seed:
+        Controls the stochastic aspects of agent behavior for reproducibility.
+
+    api_model_version:
+        Optionally specify the version of the model. If None is passed which is by default, 
+        the best model will be used.
+    
+    single_call_agent_limit:
+        The number of agents allowed in a region before it must subdivide. Currently this 
+        value represents the capacity of a quadtree leaf node that will subdivide if the 
+        number of vehicles in the region passes this threshold.
```

Comment:
```suggestion
        The number of agents allowed in a region before it must subdivide. Currently this 
        value represents the capacity of a quadtree leaf node that will subdivide if the 
        number of vehicles in the region passes this threshold. In any case,
        this will be limited to the maximum currently supported by :func:`drive`.
```
Assuming we clip this value at 100, which I think we should do.

Reply:
Done.